setlexsem/generate/generate_prompts.py

```diff
@@ -67,23 +122,136 @@ def create_prompt(
         prompt = get_prompt(
             A,
             B,
-            prompt_config,
+            prompt_config_ready,
             add_roles=add_roles,
         )
 
-        ground_truth = get_ground_truth(prompt_config.operation, A, B)
+        ground_truth = get_ground_truth(prompt_config_ready.operation, A, B)
 
         prompt_and_ground_truth.append(
             {
                 "prompt": prompt,
                 "ground_truth": ground_truth,
-                **prompt_config.to_dict(),
+                **prompt_config_ready.to_dict(),
             }
         )
 
     return prompt_and_ground_truth
 
 
+def generate_prompts(
+    set_types=None,
+    n=None,
+    m=None,
+    item_len=None,
+    decile_group=None,
+    swap_status=None,
+    overlap_fraction=None,
+    op_list=None,
+    k_shot=None,
+    prompt_type=None,
+    prompt_approach=None,
+    is_fix_shot=None,
+    number_of_data_points=10,
+    random_seed_value=292,
+    add_roles=False,
+    data_config: Dict[str, List[Union[str, int]]] = {},
+    prompt_config: Dict[str, List[Union[str, int]]] = {},
+):
+    """Generate prompts for the given hyperparameters"""
+    # generator for set construction
+    if data_config:
+        make_hps_generator = make_hps(config=data_config)
+    else:
+        make_hps_generator = make_hps(
+            config={
+                "set_types": set_types,
+                "n": n,
+                "m": m,
+                "item_len": item_len,
+                "decile_group": decile_group,
+                "swap_status": swap_status,
+                "overlap_fraction": overlap_fraction,
+            }
+        )
+
+    # generator for prompts
+    if prompt_config:
+        make_hps_generator = make_hps(config=prompt_config)
+    else:
+        make_hps_prompt_generator = make_hps_prompt(
+            config={
+                "op_list": op_list,
+                "k_shot": k_shot,
+                "prompt_type": prompt_type,
+                "prompt_approach": prompt_approach,
+                "is_fix_shot": is_fix_shot,
+            }
+        )
+
+    make_hps_generator, make_hps_generator_copy = itertools.tee(
+        make_hps_generator
+    )
+
+    # report number of overall experiments
+    n_experiments = len(list(make_hps_prompt_generator)) * len(
+        list(make_hps_generator_copy)
+    )
+    print(f"Experiment will run for {n_experiments} times")
+
+    # go through hyperparameters and run the experiment
```

Comment:
Is this comment obsolete? It seems that this function is for making prompts, not running experiments.